lib/dsmrreader.js

```diff
@@ -0,0 +1,127 @@
+                        'unit': 'kWh',
+                    },
+                    'actual': {
+                        'reading': output.power.actualProduced,
+                        'unit': 'kW',
+                    },
+                },
+                'tariffIndicator': output.power.activeTariff,
+                'switchPosition': output.power.switchPosition,
+                'voltageSags': {
+                    'L1': output.power.voltageSagsL1,
+                    'L2': output.power.voltageSagsL2,
+                    'L3': output.power.voltageSagsL3,
+                },
+                'voltageSwell': {
+                    'L1': output.power.voltageSwellsL1,
+                    'L2': output.power.voltageSwellsL2,
+                    'L3': output.power.voltageSwellsL3,
+                },
+                'instantaneous': {
+                    'current': {
+                        'L1': {
+                            'reading': output.power.instantaneousCurrentL1,
+                            'unit': 'A',
+                        },
+                        'L2': {
+                            'reading': output.power.instantaneousCurrentL2,
+                            'unit': 'A',
+                        },
+                        'L3': {
+                            'reading': output.power.instantaneousCurrentL3,
+                            'unit': 'A',
+                        },
+                    },
+                    'power': {
+                        'positive': {
+                            'L1': {
+                                'reading': output.power.instantaneousProducedElectricityL1,
+                                'unit': 'kW',
+                            },
+                            'L2': {
+                                'reading': output.power.instantaneousProducedElectricityL2,
+                                'unit': 'kW',
+                            },
+                            'L3': {
+                                'reading': output.power.instantaneousProducedElectricityL3,
+                                'unit': 'kW',
+                            },
+                        },
+                        'negative': {
+                            'L1': {
+                                'reading': output.power.instantaneousConsumedElectricityL1,
+                                'unit': 'kW',
+                            },
+                            'L2': {
+                                'reading': output.power.instantaneousConsumedElectricityL2,
+                                'unit': 'kW',
+                            },
+                            'L3': {
+                                'reading': output.power.instantaneousConsumedElectricityL3,
+                                'unit': 'kW',
+                            },
+                        },
+                    },
+                },
+            }
+        }
+
+        // backwards compatibility
+        data.gas = {
```

Comment:
This option is deprecated and not needed anymore

Reply:
I have removed:
`        // backwards compatibility
        data.gas = {
            'reading': null,
        }`